control-plane/pkg/reconciler/kafka/kafka.go

```diff
@@ -0,0 +1,219 @@
+			triggerAuth.Spec.SecretTargetRef = secretTargetRefs
+		}
+
+		if cg.Spec.Template.Spec.Auth.NetSpec.TLS.Enable {
+			secret.StringData["tls"] = "enable"
+			tls := kedav1alpha1.AuthSecretTargetRef{Parameter: "tls", Name: secret.Name, Key: "tls"}
+			secretTargetRefs = append(secretTargetRefs, tls)
+
+			if cg.Spec.Template.Spec.Auth.NetSpec.TLS.CACert.SecretKeyRef != nil {
+				ca := kedav1alpha1.AuthSecretTargetRef{
+					Parameter: "ca",
+					Name:      cg.Spec.Template.Spec.Auth.NetSpec.TLS.CACert.SecretKeyRef.Name,
+					Key:       cg.Spec.Template.Spec.Auth.NetSpec.TLS.CACert.SecretKeyRef.Key,
+				}
+
+				secretTargetRefs = append(secretTargetRefs, ca)
+			}
+
+			if cg.Spec.Template.Spec.Auth.NetSpec.TLS.Cert.SecretKeyRef != nil {
+				cert := kedav1alpha1.AuthSecretTargetRef{
+					Parameter: "cert",
+					Name:      cg.Spec.Template.Spec.Auth.NetSpec.TLS.Cert.SecretKeyRef.Name,
+					Key:       cg.Spec.Template.Spec.Auth.NetSpec.TLS.Cert.SecretKeyRef.Key,
+				}
+				secretTargetRefs = append(secretTargetRefs, cert)
+			}
+
+			if cg.Spec.Template.Spec.Auth.NetSpec.TLS.Key.SecretKeyRef != nil {
+				key := kedav1alpha1.AuthSecretTargetRef{
+					Parameter: "key",
+					Name:      cg.Spec.Template.Spec.Auth.NetSpec.TLS.Key.SecretKeyRef.Name,
+					Key:       cg.Spec.Template.Spec.Auth.NetSpec.TLS.Key.SecretKeyRef.Key,
+				}
+				secretTargetRefs = append(secretTargetRefs, key)
+			}
+			triggerAuth.Spec.SecretTargetRef = secretTargetRefs
+		}
+	}
+
+	if cg.Spec.Template.Spec.Auth.AuthSpec != nil && cg.Spec.Template.Spec.Auth.AuthSpec.Secret.Ref.Name != "" {
+		host := kedav1alpha1.AuthSecretTargetRef{
+			Parameter: "host",
+			Name:      cg.Spec.Template.Spec.Auth.AuthSpec.Secret.Ref.Name,
+			Key:       "", //TODO: add key
+		}
+		secretTargetRefs = append(secretTargetRefs, host)
+		triggerAuth.Spec.SecretTargetRef = secretTargetRefs
+		return triggerAuth, nil, nil
+	}
+
+	return triggerAuth, &secret, nil
+}
+
+// scaleObjectCreated makes a new reconciler event with event type Normal, and
+// reason ScaledObjectCreated.
+func ScaleObjectCreated(namespace, name string) reconciler.Event {
+	return reconciler.NewEvent(corev1.EventTypeNormal, "ScaledObjectCreated", "ScaledObject created: \"%s/%s\"", namespace, name)
+}
+
+// scaleObjectUpdated makes a new reconciler event with event type Normal, and
+// reason ScaledObjectUpdated.
+func ScaleObjectUpdated(namespace, name string) reconciler.Event {
+	return reconciler.NewEvent(corev1.EventTypeNormal, "ScaledObjectUpdated", "ScaledObject updated: \"%s/%s\"", namespace, name)
+}
+
+// scaleObjectDeploymentFailed makes a new reconciler event with event type Warning, and
+// reason ScaleObjectDeploymentFailed.
+func ScaleObjectDeploymentFailed(namespace, name string, err error) reconciler.Event {
+	return reconciler.NewEvent(corev1.EventTypeWarning, "ScaleObjectDeploymentFailed", "ScaledObject deployment failed to: \"%s/%s\", %w", namespace, name, err)
+}
```

Comment:
what "deployment" mean in this context?

Reply:
Ah it only meant "installation" of scaled object resource. Re-did it.